pimcore/lib/Pimcore/Document/Newsletter/AddressSourceAdapterFactoryInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Enterprise License (PEL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace Pimcore\Document\Newsletter;
+
+use Pimcore\Document\Newsletter\AddressSourceAdapterInterface;
+
+interface AddressSourceAdapterFactoryInterface
+{
+    /**
+     * Configures and creates the AddressSourceAdapterInterface
+     *
+     * @param $params
+     * @return AddressSourceAdapterInterface
+     */
+    public function configure($params);
```

Comment:
I'd prefer a name which indicates that an instance is built here (create, build, ...). `configure()` sounds more like "take an existing instance and add configuration".